test/test_c10d.py

```diff
@@ -72,6 +72,22 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def retry_on_address_already_in_use_error(func):
+    """Reruns a test if it sees "Address already in use" error."""
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        for _ in range(10):
+            try:
+                return func(*args, **kwargs)
+            except RuntimeError as error:
+                if str(error) == "Address already in use":
+                    continue
+                raise
+        raise RuntimeError("Ran out of retry attempts")
```

Comment:
Can you rephrase the error message to be more specific?

Reply:
Refactored to raise the original message.